logging/logging.go

```diff
@@ -0,0 +1,107 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	tflogging "github.com/hashicorp/terraform-plugin-sdk/helper/logging"
+	ocmlogging "github.com/openshift-online/ocm-sdk-go/logging"
+)
+
+// TfLogger is a logger that uses the Go `log` package.
+type TfLogger struct {
+	debugEnabled bool
+	infoEnabled  bool
+	warnEnabled  bool
+	errorEnabled bool
+}
+
+// New creates the provider.
+func New() ocmlogging.Logger {
+	tfLogger := &TfLogger{}
+	logLevel := tflogging.LogLevel()
+	switch logLevel {
+	case "TRACE", "DEBUG":
```

Comment:
Am I missing something or is this struct re-implements standard logging functionality?
I mean, there got to be an existing logging implementation that allows to set a log-level and write lines based on it
(also for the "panic on fatal" behavior. it should be pretty common)